gsms.py

```diff
@@ -204,6 +232,47 @@ def main() -> None:
                                 'Use the `--apps` arg to specify the full path of the file if you\'d like to use a '
                                 'custom location.')
 
+# Code from here and modified to work in this project
+# https://gist.github.com/mkropat/7550097
+class GUID(ctypes.Structure):
```

Comment:
Please add doc strings to your new classes and functions. We use numpydoc for documenting. https://numpydoc.readthedocs.io/en/latest/format.html

The classes should probably be before the functions, unless there's some conflict with them being there.

Reply:
Moved to the top and commented the classes and function